tests/scripts/basic-build-test.sh

```diff
@@ -201,9 +246,18 @@ FUNCS_PERCENT=$((1000*$FUNCS_TESTED/$FUNCS_TOTAL))
 FUNCS_PERCENT="$(($FUNCS_PERCENT/10)).$(($FUNCS_PERCENT-($FUNCS_PERCENT/10)*10))"
 
 echo "Lines Tested       : $LINES_TESTED of $LINES_TOTAL $LINES_PERCENT%"
+if [ $(echo "$LINES_PERCENT < $LINE_COV_THRESHOLD" | bc -l) = "1" ]; then
+    echo "Test error! Line coverage was below test threshold $LINE_COV_THRESHOLD"
+    EXITCODE=1
+fi
+
 echo "Functions Tested   : $FUNCS_TESTED of $FUNCS_TOTAL $FUNCS_PERCENT%"
-echo
+if [ $(echo "$FUNCS_PERCENT < $FN_COV_THRESHOLD" | bc -l) = "1" ]; then
```

Comment:
Minor: Is the `-l` option needed here for bc?

Reply:
Just to clarify: bc supports floats of arbitrary precision without the `-l` options, the problem is that the `scale` (the number of decimal digits after the point) is set to 0 by default. The effect of the `-l` option is to include the standard math libraries (which IMHO are not needed in this case) and to set the scale to 20 (according to POSIX at least). The script would work just as well if bc was also told to execute the statement `scale=20;` before the float arithmetic to set the precision to whatever is needed.

If the scale differences across different platforms is a concern, then I would recommend to manually set the precision manually (regardless of the `-l` option) so that we do not get slightly different results in different settings due to this problem...

Reply:
I think we're looking at different documentation for `bc`, but now I look deeper, I can see this won't work for POSIX anyway, because apparently POSIX `bc` can't evaluate expressions outside of an `if`. I've removed the `-l` option, and given `basic-build-test.sh` only works on Linux today anyway, I guess this is good enough.